odl/test/solvers/nonsmooth/adupdates_test.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Test for the Fessler alternating dual updates solver."""
+
+from __future__ import division
+# import pytest
+import odl
+# import numpy as np
+from odl.solvers.nonsmooth import adupdates, adupdates_simple
+
+from odl.util.testutils import all_almost_equal
+
+
+# Places for the accepted error when comparing results
+HIGH_ACCURACY = 8
+LOW_ACCURACY = 4
+
+
+def test_adupdates():
+    """Test if the adupdates solver handles the following problem correctly:
+
+    ( 1  1/2 1/3 1/4) (x_1)   (a)
+    (1/2 1/3 1/4 1/5) (x_2)   (b)
+    (1/3 1/4 1/5 1/6) (x_3) = (c)
+    (1/4 1/5 1/6 1/7) (x_4)   (d).
+
+    The matrix is the ill-conditined Hilbert matrix, the inverse of which
+    can be given in closed form. If we set
+
+    (a)   (25/12)                    (x_1) = (1)
+    (b)   (77/60)                    (x_2) = (1)
+    (c) = (19/20)         then       (x_3) = (1)
+    (d)   (319/420),                 (x_4) = (1).
+
+    We therefore solve the problem
+
+    min ||Ax - b||^2 + TV(x) s.t. x >= 0
```

Comment:
I was fine until this point. Why should this have the same solution as above? What does TV even mean here?